src/firefly/js/visualize/ui/MouseReadout.jsx

```diff
@@ -30,75 +33,249 @@ var rS= {
+	//const hideReadout= renderEmpty(plotView,plot, mouseState, visRoot);
+
+	if (!magMouse.includes(mouseState.mouseState)) EMPTY; //hideReadout;
+
+	var spt= mouseState.screenPt;
+	if (!spt) return EMPTY;//hideReadout;
 
-	if (!pv || !mouseState) return EMPTY;
 
-	var plot= primePlot(pv);
 
-	var leftColumn = {width: 200, display: 'inline-block'};
+	var leftColumn = {width: 140, display: 'inline-block'};
 
-	var rightColumn = {display: 'inline-block'};
+	var title = plotView.plots[0].title;
+	var middleColumn = {width: 230, display: 'inline-block'};
 	var  textStyle = {textDecoration: 'underline', color: 'DarkGray', fontStyle:'italic' ,  display: 'inline-block'};
+	var rightColumn = {paddingLeft: '35px',  display: 'inline-block'};
 	return (
 			<div style={ rS}>
                <div>
+				  <div style={leftColumn} onClick={ () => showDialog('pixelSize', visRoot.pixelSize)}>
+					  <div style={ textStyle} > {getLabel(visRoot.pixelSize) }</div>
+					  { showReadout(plot, mouseState,visRoot.pixelSize)}
+				  </div>
 
-				 <div	style={leftColumn} onClick={ () => showDialog('pixelSize')}>
-					 <div style={ textStyle} > { updateField('pixelSize')}</div>
-				 </div>
-
-				 <div   style={rightColumn} onClick={ () => showDialog('coordinateSys' )}>
-					 <div style={ textStyle} >{ updateField('coordinateSys')} </div>
-					 {showReadout(plot, mouseState, CoordinateSys.EQ_J2000)}</div>
+				  <div style={middleColumn} onClick={ () => showDialog('readout1' ,visRoot.mouseReadout1)}>
+					 <div style={ textStyle} > { getLabel( visRoot.mouseReadout1) } </div>
+					 {showReadout(plot, mouseState,visRoot.mouseReadout1)}
+				  </div>
 
+				  <div style={rightColumn}> {title}  </div>
               </div>
-	         <div>
-				 <div	style={leftColumn} > {showReadout(plot, mouseState) } </div>
-				 <div style={ rightColumn}  onClick={ () => showDialog('imagePixel' )}>
-					 <div style={ textStyle} >{updateField('imagePixel' )} </div>
-					 {showReadout(plot, mouseState, CoordinateSys.PIXEL)}
-				 </div>
-		    </div>
+
+			  <div>
+				  <div style={leftColumn} > {showReadout(plot, mouseState,visRoot.flux ) } </div>
+				  <div style={ middleColumn}  onClick={ () => showDialog('readout2' ,visRoot.mouseReadout2)}>
+					 <div style={ textStyle} >{getLabel( visRoot.mouseReadout2)} </div>
+					 {showReadout(plot, mouseState, visRoot.mouseReadout2)}
+				  </div>
+				  <div style={rightColumn} title='Click on an image to lock the display at that point.'   >
+					  <input type='checkbox' name='aLock' value='lock'
+							 onChange = { (request) => setClickLock(plot,mouseState , request) } />
+					  Lock by click
+				  </div>
+		      </div>
 
 		  </div>
 
 	);
 }
 
-function showReadout(plot, mouseState, coordinate){
-	if (!plot) return'';
-	if (isBlankImage(plot)) return'';
+MouseReadout.propTypes= {
+	visRoot:React.PropTypes.object.isRequired,
+	plotView: React.PropTypes.object,
+	mouseState:React.PropTypes.object.isRequired,
+};
+
```

Comment:
You can import PropTypes directly so you don't have to use React.PropTypes.
`import React, {PropTypes} from 'react;`